### PR TITLE
Onboard app level pvtkeyjwt reuse config

### PR DESCRIPTION
**Pull request by mpmadhavig**
- `components/org.wso2.carbon.identity.api.server.dcr`
    - Updated the DTOs related to DCR endpoint.

- `components/org.wso2.carbon.identity.oauth.common`
    - Constant re-initializing to avoid the cyclic dependency.

- `components/org.wso2.carbon.identity.oauth.dcr`
    - Updated the beans related DCR endpoint.

- `components/org.wso2.carbon.identity.oauth`
    - Updated DTOs and beans related OAuth Apps.
    - Implemented logic to read org level config using the `ConfigurationManager`.

### Please Note:

- The config will be saved as a SP OIDC property. 
- For the already created applications, as the config is not added, the old org level config will be taken.
    - When retrieving the application via a GET request, if the config is unavailable, org level config will be set.
- For the newly created applications, the user configured value will be saved.
    - The org level config will be totally irrelevant for the new application.

Related PRs:
> https://github.com/wso2/product-is/issues/20546

### CODE REVIEW

**Review by shashimalcse — APPROVED** [review]

**Review by shashimalcse — DISMISSED** [review]

**Review by shashimalcse — APPROVED** [review]

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java [hunk 1]
```diff
@@ -429,6 +430,13 @@ OAuthConsumerAppDTO registerAndRetrieveOAuthApplicationData(OAuthConsumerAppDTO
                             }
                             app.setTokenEndpointAuthMethod(tokenEndpointAuthMethod);
                         }
+                        Boolean tokenEndpointAllowReusePvtKeyJwt = application.isTokenEndpointAllowReusePvtKeyJwt();
+                        if (isInvalidTokenEPReusePvtKeyJwtRequest(tokenEndpointAuthMethod,
+                                tokenEndpointAllowReusePvtKeyJwt)) {
+                            throw handleClientError(INVALID_REQUEST,
+                                    "Invalid token endpoint authentication method requested.");
```
Comment (Thumimku) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java:437]: Let's add much more detail error description.

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java [hunk 2]
```diff
@@ -2492,6 +2507,27 @@ private void handleInternalTokenRevocation(String consumerKey, Properties proper
...
+    private boolean isInvalidTokenEPReusePvtKeyJwtRequest(String tokenEndpointAuthMethod,
+                                                          Boolean tokenEndpointAllowReusePvtKeyJwt) {
+
+        if (StringUtils.isNotBlank(tokenEndpointAuthMethod)) {
+            if (tokenEndpointAuthMethod.equals(PRIVATE_KEY_JWT)) {
+                return tokenEndpointAllowReusePvtKeyJwt == null;
+            } else {
+                return tokenEndpointAllowReusePvtKeyJwt != null;
+            }
+        } else {
```
Comment (Thumimku) [inline components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java:2526]: we don't need this else block.
```
    if (StringUtils.isBlank(tokenEndpointAuthMethod)) {
        return tokenEndpointAllowReusePvtKeyJwt != null;
    }

    if (tokenEndpointAuthMethod.equals(PRIVATE_KEY_JWT)) {
        return tokenEndpointAllowReusePvtKeyJwt == null;
    }

    return tokenEndpointAllowReusePvtKeyJwt != null;
```

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java (hunk 1 above)
Comment (mpmadhavig) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java]: Addressed with 1346c762ac6987102496807d3d874923edeb3003

File: components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java (hunk 2 above)
Comment (mpmadhavig) [reply components/org.wso2.carbon.identity.oauth/src/main/java/org/wso2/carbon/identity/oauth/OAuthAdminServiceImpl.java]: Addressed with 1346c762ac6987102496807d3d874923edeb3003

**Pull request merged**